### DISCUSSION TITLE
The way withBadge is displayed.

### DISCUSSION THREAD

**Discussion by philipbaginski** [laravel/nova-issues#4322] [Q&A]
OS | Windows NT (10.0 - build 18363 (Windows 10))
PHP Version | 8.0.3
Database | 5.7.24
Laravel Version | 9.14.1
Nova Version | 4.7.0 (Silver Surfer)

Description:

In menu section we can add withBadge trait and it will show the number (count method) of models based on the conditions we apply.

But what about situation we have no models to display following the conditions? We still see badge and number 0 displayed.

Is there any possibility to hide withBadge when there is nothing to show?

**Comment by crynobone** [comment]
You can do the following:

```php
use Laravel\Nova\Badge;

Badge::$types['hidden'] = 'hidden';

// and assign `hidden` when badge value is 0.
return Badge::make(0, 'hidden');
```

**Reply by davidhemphill** [reply]
I would not recommend this method. While this works at the moment, the `$type` parameter for `Badge` may not necessarily map 1:1 to CSS classes in the future. 

Right now there is no way to conditionally display the badge.

**Reply by davidhemphill** [reply]
Just added a `withBadgeIf` method to Nova. It'll be in the next release.
